array.go

```diff
@@ -360,6 +360,97 @@ func (a *Array) NonEmptyDomain() ([]NonEmptyDomain, bool, error) {
 	}
 }
 
+// NonEmptyDomainVarFromName retrieves the non-empty domain from an array for a
+// given dimension name.
+// This returns the bounding coordinates for the dimension
+func (a *Array) NonEmptyDomainVarFromName(dimName string) ([]NonEmptyDomain, []NonEmptyDomain, bool, error) {
+
+	schema, err := a.Schema()
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	domain, err := schema.Domain()
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	domainType, err := domain.Type()
```

Comment:
Domain type can not be used here. We should no longer use domain type anywhere anywhere. Its is only valid for homogeneous dimensions. We shouldn't even try to use it though because using the specific dimension datatype will work in all cases.

Reply:
Got it